src/AutoRest.CSharp/Common/Generation/Types/TypeFactory.cs

```diff
@@ -103,6 +103,58 @@ public TypeFactory(OutputLibrary library)
             _ => _library.FindTypeForSchema(schema).WithNullable(isNullable)
         };
 
+        /// <summary>
+        /// Similar to <see cref="CreateType(InputType)"/>, this method creates a CSharpType from an InputType that is not a model type.
+        /// If the input type is a model type, this method will throw an exception.
+        /// </summary>
+        /// <param name="inputType">the input type to convert.</param>
+        /// <returns>the converted <c>CSharpType</c>.</returns>
+        public static CSharpType CreateSimpleType(InputType inputType) => inputType switch
```

Comment:
we should not duplicate the logic between these, lets find a way to share it

Reply:
fixed.